src/monitor.rs

```diff
@@ -448,11 +456,13 @@ async fn submit_and_watch_solution<T: MinerConfig + Send + Sync + 'static>(
 						.find_first::<runtime::election_provider_multi_phase::events::SolutionStored>(
 						);
 
-				return if let Ok(Some(_)) = solution_stored {
+				if let Ok(Some(_)) = solution_stored {
 					log::info!(target: LOG_TARGET, "Included at {:?}", details.block_hash());
-					Ok(())
+					if let Listen::Head = listen {
```

Comment:
if one "--listen head` it's fine to regard it as completed here but not in when `--listen finalized`

this is done to avoid submitting duplicate solutions by mistake